"Manifest generation failed" error in SBOM generation for all build legs

The Generate SBoM Manifest step from the 1ES templates is failing all build legs (excluding the ones with guardian failures) with the following error:

```
##[error]Encountered an error while generating the manifest.
##[error]Error details: Access to the path '/mnt/vss/_work/1/s/artifacts/staging/_manifest' is denied.
```
This step is "succeeding with errors" so it's not failing the overall build. This occurs for 10.0 VMR, but it does not occur in builds of 9.0 and 8.0.

This is likely a regression caused by https://github.com/dotnet/sdk/pull/45323

Hi @ellahathaway @ViktorHofer , I reopened this issue because the offline legs are still failed with the same issue:

    ##[error]Encountered an error while generating the manifest.
    ##[error]Error details: Access to the path '/__w/1/a/artifacts/_manifest' is denied.
[Example build](https://dev.azure.com/dnceng/internal/_build/results?buildId=2600908&view=logs&j=391e3a46-1d0e-50f6-4b48-986aa9de4d73&t=e9fc764c-5d81-571b-5997-536fd39a4e99) (internal Microsoft link)

--- a/eng/pipelines/templates/jobs/vmr-build.yml
+++ b/eng/pipelines/templates/jobs/vmr-build.yml
@@ -432,6 +432,13 @@ jobs:
       displayName: Build
       workingDirectory: $(sourcesPath)
 
+    - ${{ if ne(parameters.runOnline, 'True' )}}:
+      - script: |
+          set -ex
+          # Update the owner of the staging directory to the current user
+          sudo chown -R $(whoami) $(artifactsStagingDir)
+        displayName: Update owner of artifacts staging directory
+
     # Only run tests if enabled
     - ${{ if eq(parameters.runTests, 'True') }}:
       # Setup the NuGet sources used by the tests to use private feeds. This is necessary when testing internal-only product

